### ISSUE TITLE
Can the log method only log the format error instead of throw the exception?

### ISSUE THREAD

**Issue by lsxia** [MengRao/fmtlog#87]
It is not that uncommon for developers to miss a "{}" in for format string, which caused format_error exception at runtime. The malformed line logging code may crash the entire application, which is not desirable. Should we only log the format_error and the line number that has the format error without crashing the application?

**Comment by MengRao** [comment]
I believe missing a {} would not throw an exception, it just leave the additonal parameters not handled, not a big issue right?

**Comment by lsxia** [comment]
No, I believe the missing {} will cause runtime failure at this moment (which already happened a few times in our system). I agree with you that it would be better than crash if just ignoring the additional parameters.
